mm2src/coins/z_coin/z_params/mod.rs

```diff
@@ -0,0 +1,85 @@
+mod indexeddb;
+pub use indexeddb::ZcashParamsWasmImpl;
+
+use blake2b_simd::State;
+use common::log::info;
+use common::log::wasm_log::register_wasm_log;
+use mm2_err_handle::prelude::MmResult;
+use mm2_err_handle::prelude::*;
+use mm2_net::wasm::http::FetchRequest;
+use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+use wasm_bindgen_test::*;
+
+const DOWNLOAD_URL: &str = "https://komodoplatform.com/downloads";
+const SAPLING_SPEND_NAME: &str = "sapling-spend.params";
+const SAPLING_OUTPUT_NAME: &str = "sapling-output.params";
+const SAPLING_SPEND_HASH: &str = "8270785a1a0d0bc77196f000ee6d221c9c9894f55307bd9357c3f0105d31ca63991ab91324160d8f53e2bbd3c2633a6eb8bdf5205d822e7f3f73edac51b2b70c";
+const SAPLING_OUTPUT_HASH: &str = "657e3d38dbb5cb5e7dd2970e8b03d69b4787dd907285b5a7f0790dcc8072f60bf593b32cc2d1c030e00ff5ae64bf84c5c3beb84ddc841d48264b4a171744d028";
+
+#[derive(Debug, derive_more::Display)]
+pub enum ZcashParamsBytesError {
+    IO(String),
+}
+
+/// Download, validate and return z_params from given `DOWNLOAD_URL`
+async fn fetch_params(name: &str, expected_hash: &str) -> MmResult<Vec<u8>, ZcashParamsBytesError> {
+    let (status, file) = FetchRequest::get(&format!("{DOWNLOAD_URL}/{name}"))
+        .cors()
```

Comment:
we dont use `cors()` for `FetchRequest::get`. Did you test it without cors?

Reply:
doesn't work too.

Reply:
Tests fail without cors call?

Reply:
cors() doesn't fix cors issues with the server until the server was updated to allow all origin..so there's no reason to use it.

Reply:
I think we should remove cors() then, if it doesnt change the situation.
Plus, I also used cors() in get req long time ago for nft, and GUI got some error on their side, after cors deletion error has gone.
As far as I remember there is no need to add cors in get req, browser does it itself.

Reply:
@borngraced  I have a question, did you check `fetch_params` process with [react-komodefi-wasm](https://github.com/KomodoPlatform/react-komodefi-wasm) project? Everything was fine in logs?

Reply:
No, I didn't test with `[react-komodefi-wasm](https://github.com/KomodoPlatform/react-komodefi-wasm)` yet. But I'll do that today.

Reply:
seems to be working as supposed 

<img width="1524" alt="Screenshot 2023-12-05 at 12 02 09" src="https://github.com/KomodoPlatform/komodo-defi-framework/assets/51881311/f527034c-1c3e-4b4b-af63-5f9488553509">

Reply:
Ok, if no cursor issues in console, then fine